[FLINK-23206] Minor tickets are moved to Not a Priority instead of cl…

knaufk opened the pull request:
…osing

knaufk committed:
[FLINK-23206] Minor tickets are moved to Not a Priority instead of closing

knaufk commented:
Once this is merged I will do a bulk change in Jira that
* moves all tickets resolution "Auto-Closed" to "Open" with priority "Not a Priority"
* replace label "auto-closed" with "auto-deprioritized-minor"

pnowojski commented on flink_jira_bot.py:
```diff
@@ -65,14 +64,16 @@ def get_args():
     )
-    stale_minor_rule = StaleMinorRule(jira, jira_bot_config["stale_minor"], args.dryrun)
-    stale_major_rule = StaleMajorOrAboveRule(
+    stale_minor_rule = StaleUnassignedRule(
+        jira, jira_bot_config["stale_minor"], args.dryrun, "Major", "Not a Priority"
+    )
+    stale_major_rule = StaleUnassignedRule(
         jira, jira_bot_config["stale_major"], args.dryrun, "Major", "Minor"
     )
```
How is it supposed to be working? We have now two `StaleUnassignedRule`s, both from `Major` one to `Minor` other to `Not a Priority` at the same time?

knaufk replied:
It is not ;) One of the them should be Minor. Fixed.

pnowojski commented on stale_unassigned_rule.py:
```diff
@@ -20,7 +20,7 @@
-class StaleMajorOrAboveRule(FlinkJiraRule):
+class StaleUnassignedRule(FlinkJiraRule):
     """
     Tickets major and above without a fixVersion need an assignee, or an update within {stale_<blocker|critical|major>.stale_days},
```
nit: outdated comment `Tickets major and above`.

knaufk replied:
Fixed.

pnowojski reviewed (commented):
I might not be the best person to review this code (first time seeing it 😳 ) but I've left two comments.

knaufk commented:
@pnowojski Thanks you for the review. Good to see that more people are familiarizing themselves with the code increasing the bus factor. I addressed your comments.

pnowojski reviewed (approved):
I'm not 100% sure but I think it LGTM :)
